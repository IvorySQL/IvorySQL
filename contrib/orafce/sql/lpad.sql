--Test For lpad
set search_path to oracle, pg_catalog;
set datestyle to ISO,YMD;
SELECT oracle.lpad(123, 20, 0);
SELECT oracle.lpad(123, 20);
SELECT oracle.lpad(123::int2, 20::int2, 0::int2);
SELECT oracle.lpad(123::int4, 20::int4, 0::int4);
SELECT oracle.lpad(123::int4, 20::int4);
SELECT oracle.lpad(123::int8, 20::int8, 0::int8);
SELECT oracle.lpad(123.99::numeric, 20.99::numeric, 0.9::numeric);
SELECT oracle.lpad(123.00::numeric, 20.99::numeric, 0.9::numeric);
SELECT oracle.lpad(123.00::numeric, 20::int, 0::int);
SELECT oracle.lpad(123.00::int, 20.99::numeric, 0.9::numeric);
SELECT oracle.lpad(123.99::numeric, 20.99::numeric);

SELECT oracle.lpad('123', '20'::int, '0');
SELECT oracle.lpad('123', '20.94'::numeric, '0');
SELECT oracle.lpad('123', '20'::int);
SELECT oracle.lpad('123', '20.95'::numeric);
SELECT oracle.lpad('I like'::char(10), 20::int, '#$'::char);
SELECT oracle.lpad('I like'::char(10), 20::int, '#$'::char(2));
SELECT oracle.lpad('I like'::char(10), 20::int);
SELECT oracle.lpad('$@'::char(2), 20::int, 'I like'::char(10));
SELECT oracle.lpad('$@'::char(2), 20.99::numeric);
SELECT oracle.lpad('I like'::varchar(20), 20::int, '#$'::varchar(20));
SELECT oracle.lpad('I like'::varchar2(20), 20::int, '#$'::varchar2(20));
SELECT oracle.lpad('I like'::nvarchar2(20), 20::int, '#$'::nvarchar2(20));
SELECT lpad('I like'::text, 20::int, '#$'::text);
SELECT oracle.lpad('I like'::text, 20.99::numeric, '#$'::text);
SELECT oracle.lpad('I like'::text, 20.99::numeric);
SELECT lpad('I Love China'::text, 25::int);
SELECT oracle.lpad('I Love China'::text, 25.99::numeric);
SELECT lpad('I Love China'::text, 7::int, '#$'::text);
SELECT oracle.lpad('I Love China'::text, 7.99::numeric, '#$'::text);
SELECT lpad('我来自China'::text, 25::int, '#$'::text);
SELECT oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text);
SELECT oracle.lpad('我来自China'::text, 25.99::numeric);
SELECT lengthb(oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text));
SELECT lengthb(oracle.lpad('我来自China'::text, 25.99::numeric));
SELECT length(oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text));
SELECT length(oracle.lpad('我来自China'::text, 25.99::numeric));

SELECT oracle.lpad('2019-12-12'::date, 20::int, '2019-12-13'::date);
SELECT oracle.lpad('2019-12-12'::date, 20.99::numeric, '2019-12-13'::date);
SELECT oracle.lpad('2019-12-12'::date, 20.99::numeric);
SELECT oracle.lpad('2019-12-12'::timestamp, 20::int, '2019-12-13'::timestamp);
SELECT oracle.lpad('2019-12-12'::timestamp, 20.99::numeric, '2019-12-13'::timestamp);
SELECT oracle.lpad('2019-12-12'::timestamp, 20.99::numeric);
SELECT oracle.lpad('2019-12-12'::date, 20::int, '2019-12-13'::timestamp);

SELECT oracle.lpad(null,null,null);
SELECT oracle.lpad(null,null);

SELECT lpad('ab'::text,-8::int,'*@'::text);
SELECT lpad('ab'::text,-1::int,'*@'::text);
SELECT oracle.lpad('ab'::text,-1.9::numeric,'*@'::text);
SELECT oracle.lpad('ab'::text,-0.9::numeric,'*@'::text);
SELECT oracle.lpad('ab'::text,0.9::numeric,'*@'::text);
SELECT lpad('ab'::text,0::int,'*@'::text);

SELECT lpad(null,2::int);
SELECT oracle.lpad(null,2.99::numeric);

SELECT lpad('ab'::text,-8::int);
SELECT lpad('ab'::text,-1::int);
SELECT oracle.lpad('ab'::text,-1.9::numeric);
SELECT oracle.lpad('ab'::text,-0.9::numeric);
SELECT oracle.lpad('ab'::text,0.9::numeric);
SELECT lpad('ab'::text,0::int);

SELECT oracle.lpad('',3::int,'');
SELECT oracle.lpad('',3.99::numeric,'');
SELECT oracle.lpad('',-3::int,'');
SELECT oracle.lpad('',0::int,'');
SELECT oracle.lpad('',3::int);
SELECT oracle.lpad('',3.99::numeric);
SELECT oracle.lpad('',-3::int);
SELECT oracle.lpad('',0::int);

SELECT oracle.lpad(' ',3::int,' ');
SELECT oracle.lpad(' ',3.99::numeric,' ');
SELECT oracle.lpad(' ',-3::int,' ');
SELECT oracle.lpad(' ',0::int,' ');
SELECT oracle.lpad(' ',3::int);
SELECT oracle.lpad(' ',3.99::numeric);
SELECT oracle.lpad(' ',-3::int);
SELECT oracle.lpad(' ',0::int);

reset search_path;