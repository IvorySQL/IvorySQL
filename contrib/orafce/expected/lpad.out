--Test For lpad
set search_path to oracle, pg_catalog;
set datestyle to ISO,YMD;
SELECT oracle.lpad(123, 20, 0);
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad(123, 20);
         lpad         
----------------------
                  123
(1 row)

SELECT oracle.lpad(123::int2, 20::int2, 0::int2);
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad(123::int4, 20::int4, 0::int4);
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad(123::int4, 20::int4);
         lpad         
----------------------
                  123
(1 row)

SELECT oracle.lpad(123::int8, 20::int8, 0::int8);
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad(123.99::numeric, 20.99::numeric, 0.9::numeric);
         lpad         
----------------------
 0.90.90.90.90.123.99
(1 row)

SELECT oracle.lpad(123.00::numeric, 20.99::numeric, 0.9::numeric);
         lpad         
----------------------
 0.90.90.90.90.123.00
(1 row)

SELECT oracle.lpad(123.00::numeric, 20::int, 0::int);
         lpad         
----------------------
 00000000000000123.00
(1 row)

SELECT oracle.lpad(123.00::int, 20.99::numeric, 0.9::numeric);
         lpad         
----------------------
 0.90.90.90.90.90.123
(1 row)

SELECT oracle.lpad(123.99::numeric, 20.99::numeric);
         lpad         
----------------------
               123.99
(1 row)

SELECT oracle.lpad('123', '20'::int, '0');
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad('123', '20.94'::numeric, '0');
         lpad         
----------------------
 00000000000000000123
(1 row)

SELECT oracle.lpad('123', '20'::int);
         lpad         
----------------------
                  123
(1 row)

SELECT oracle.lpad('123', '20.95'::numeric);
         lpad         
----------------------
                  123
(1 row)

SELECT oracle.lpad('I like'::char(10), 20::int, '#$'::char);
         lpad         
----------------------
 ##########I like    
(1 row)

SELECT oracle.lpad('I like'::char(10), 20::int, '#$'::char(2));
         lpad         
----------------------
 #$#$#$#$#$I like    
(1 row)

SELECT oracle.lpad('I like'::char(10), 20::int);
         lpad         
----------------------
           I like    
(1 row)

SELECT oracle.lpad('$@'::char(2), 20::int, 'I like'::char(10));
         lpad         
----------------------
 I like    I like  $@
(1 row)

SELECT oracle.lpad('$@'::char(2), 20.99::numeric);
         lpad         
----------------------
                   $@
(1 row)

SELECT oracle.lpad('I like'::varchar(20), 20::int, '#$'::varchar(20));
         lpad         
----------------------
 #$#$#$#$#$#$#$I like
(1 row)

SELECT oracle.lpad('I like'::varchar2(20), 20::int, '#$'::varchar2(20));
         lpad         
----------------------
 #$#$#$#$#$#$#$I like
(1 row)

SELECT oracle.lpad('I like'::nvarchar2(20), 20::int, '#$'::nvarchar2(20));
         lpad         
----------------------
 #$#$#$#$#$#$#$I like
(1 row)

SELECT lpad('I like'::text, 20::int, '#$'::text);
         lpad         
----------------------
 #$#$#$#$#$#$#$I like
(1 row)

SELECT oracle.lpad('I like'::text, 20.99::numeric, '#$'::text);
         lpad         
----------------------
 #$#$#$#$#$#$#$I like
(1 row)

SELECT oracle.lpad('I like'::text, 20.99::numeric);
         lpad         
----------------------
               I like
(1 row)

SELECT lpad('I Love China'::text, 25::int);
           lpad            
---------------------------
              I Love China
(1 row)

SELECT oracle.lpad('I Love China'::text, 25.99::numeric);
           lpad            
---------------------------
              I Love China
(1 row)

SELECT lpad('I Love China'::text, 7::int, '#$'::text);
  lpad   
---------
 I Love 
(1 row)

SELECT oracle.lpad('I Love China'::text, 7.99::numeric, '#$'::text);
  lpad   
---------
 I Love 
(1 row)

SELECT lpad('我来自China'::text, 25::int, '#$'::text);
           lpad            
---------------------------
 #$#$#$#$#$#$#$我来自China
(1 row)

SELECT oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text);
           lpad            
---------------------------
 #$#$#$#$#$#$#$我来自China
(1 row)

SELECT oracle.lpad('我来自China'::text, 25.99::numeric);
           lpad            
---------------------------
               我来自China
(1 row)

SELECT lengthb(oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text));
 lengthb 
---------
      28
(1 row)

SELECT lengthb(oracle.lpad('我来自China'::text, 25.99::numeric));
 lengthb 
---------
      28
(1 row)

SELECT length(oracle.lpad('我来自China'::text, 25.99::numeric, '#$'::text));
 length 
--------
     22
(1 row)

SELECT length(oracle.lpad('我来自China'::text, 25.99::numeric));
 length 
--------
     22
(1 row)

SELECT oracle.lpad('2019-12-12'::date, 20::int, '2019-12-13'::date);
         lpad         
----------------------
 22019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::date, 20.99::numeric, '2019-12-13'::date);
         lpad         
----------------------
 22019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::date, 20.99::numeric);
         lpad         
----------------------
  2019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::timestamp, 20::int, '2019-12-13'::timestamp);
         lpad         
----------------------
 22019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::timestamp, 20.99::numeric, '2019-12-13'::timestamp);
         lpad         
----------------------
 22019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::timestamp, 20.99::numeric);
         lpad         
----------------------
  2019-12-12 00:00:00
(1 row)

SELECT oracle.lpad('2019-12-12'::date, 20::int, '2019-12-13'::timestamp);
         lpad         
----------------------
 22019-12-12 00:00:00
(1 row)

SELECT oracle.lpad(null,null,null);
 lpad 
------
 
(1 row)

SELECT oracle.lpad(null,null);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,-8::int,'*@'::text);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,-1::int,'*@'::text);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,-1.9::numeric,'*@'::text);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,-0.9::numeric,'*@'::text);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,0.9::numeric,'*@'::text);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,0::int,'*@'::text);
 lpad 
------
 
(1 row)

SELECT lpad(null,2::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad(null,2.99::numeric);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,-8::int);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,-1::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,-1.9::numeric);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,-0.9::numeric);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('ab'::text,0.9::numeric);
 lpad 
------
 
(1 row)

SELECT lpad('ab'::text,0::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',3::int,'');
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',3.99::numeric,'');
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',-3::int,'');
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',0::int,'');
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',3::int);
 lpad 
------
    
(1 row)

SELECT oracle.lpad('',3.99::numeric);
 lpad 
------
    
(1 row)

SELECT oracle.lpad('',-3::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad('',0::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad(' ',3::int,' ');
 lpad 
------
    
(1 row)

SELECT oracle.lpad(' ',3.99::numeric,' ');
 lpad 
------
    
(1 row)

SELECT oracle.lpad(' ',-3::int,' ');
 lpad 
------
 
(1 row)

SELECT oracle.lpad(' ',0::int,' ');
 lpad 
------
 
(1 row)

SELECT oracle.lpad(' ',3::int);
 lpad 
------
    
(1 row)

SELECT oracle.lpad(' ',3.99::numeric);
 lpad 
------
    
(1 row)

SELECT oracle.lpad(' ',-3::int);
 lpad 
------
 
(1 row)

SELECT oracle.lpad(' ',0::int);
 lpad 
------
 
(1 row)

reset search_path;
